client/src/components/Collections/common/CollectionEditView.vue

```diff
@@ -0,0 +1,225 @@
+            return _l(str);
+        },
+        getDatatypesAndGenomes: async function () {
+            let datatypes = store.getters.getUploadDatatypes();
+            if (!datatypes || datatypes.length == 0) {
+                await store.dispatch("fetchUploadDatatypes");
+                datatypes = store.getters.getUploadDatatypes();
+            }
+            this.extensions = datatypes;
+            let genomes = store.getters.getUploadGenomes();
+            if (!genomes || genomes.length == 0) {
+                await store.dispatch("fetchUploadGenomes");
+                genomes = store.getters.getUploadGenomes();
+            }
+            this.genomes = genomes;
+        },
+        getCollectionDataAndAttributes: async function () {
+            let attributesGet = store.getters.getCollectionAttributes(this.collection_id);
+            if (attributesGet == null) {
+                await store.dispatch("fetchCollectionAttributes", this.collection_id);
+                attributesGet = store.getters.getCollectionAttributes(this.collection_id);
+            }
+            this.attributes_data = attributesGet;
+            this.getDatabaseKeyFromElements();
+            this.getExtensionFromElements();
+        },
+        getDatabaseKeyFromElements: function () {
+            this.databaseKeyFromElements = this.attributes_data.dbkey;
+        },
+        getExtensionFromElements: function () {
+            this.datatypeFromElements = this.attributes_data.extension;
+            this.selectedExtension = this.extensions.find((element) => element.id == this.datatypeFromElements);
+        },
+        getConverterList: async function () {
+            axios.get(prependPath("/api/datatypes/suitable_converters/" + this.collection_id)).then((response) => {
+                this.suitableConverters = response.data;
+            });
+        },
+        clickedSave: function (attribute, newValue) {
+            const url = prependPath("/api/dataset_collections/" + this.collection_id);
+            const data = {};
+            if (attribute == "dbkey") {
+                data["dbkey"] = newValue.id;
+            } else if (attribute == "file_ext") {
+                data["file_ext"] = newValue.id;
+            }
+            axios
+                .put(url, data)
+                .then((response) => {
+                    this.apiCallToGetData();
+                    this.getDatabaseKeyFromElements();
+                    this.getExtensionFromElements();
+                })
+                .catch(this.handleError);
+        },
+        clickedConvert: function () {
+            console.log("this.chosenConverter = ", this.chosenConverter);
+            const url = prependPath("/api/tools/");
+            const data = {
+                tool_id: this.chosenConverter.tool_id,
+                inputs: { input: { batch: true, values: [{ src: "hdca", id: this.collection_id }] } },
```

Comment:
That is probably not going to work for all converters. It's not necessary that the data input parameter name is `input`, some of them are `input1` (for instance, CONVERTER_bz2_to_uncompressed, CONVERTER_gz_to_uncompressed and many more) and we don't have control over all of the converters.
It's worth checking if it works, maybe we have some magic for converters in place to make this work, but I don't think so.

We could work around that by adding an API endpoint that just takes a collection and the target datatype and that calls into the tool submission.

Reply:
@mvdbeek Am I correct in saying that we discovered that there is some magic in place?